Source/Microsoft.Teams.Apps.CompanyCommunicator.Send.Func/Services/DataServices/ManageResultDataService.cs

```diff
@@ -0,0 +1,141 @@
+            {
+                allStatusCodeResults = $"{existingSentNotificationDataEntity.AllStatusCodeResults}{statusCode.ToString()},";
+                numberOfAttemptsToSend = existingSentNotificationDataEntity.NumberOfAttemptsToSend + 1;
+
+                // Do not send message to data queue in order to not multi-count messages to users
+                sendDataQueueMessage = false;
+            }
+
+            var updatedSentNotificationDataEntity = new SentNotificationDataEntity
+            {
+                PartitionKey = notificationId,
+                RowKey = aadId,
+                AadId = aadId,
+                TotalNumberOfThrottles = totalNumberOfThrottles,
+                SentDate = currentDateTimeUtc,
+                IsStatusCodeFromCreateConversation = isStatusCodeFromCreateConversation,
+                StatusCode = (int)statusCode,
+                ErrorMessage = errorMessage,
+                AllStatusCodeResults = allStatusCodeResults,
+                NumberOfAttemptsToSend = numberOfAttemptsToSend,
+            };
+
+            if (statusCode == HttpStatusCode.Created)
+            {
+                updatedSentNotificationDataEntity.DeliveryStatus = SentNotificationDataEntity.Succeeded;
+                dataQueueMessageContent.ResultType = DataQueueResultType.Succeeded;
+            }
+            else if (statusCode == HttpStatusCode.TooManyRequests)
+            {
+                updatedSentNotificationDataEntity.DeliveryStatus = SentNotificationDataEntity.Throttled;
+                dataQueueMessageContent.ResultType = DataQueueResultType.Throttled;
+            }
+            else if (statusCode == HttpStatusCode.Continue)
+            {
+                // This is a special case where an exception was thrown in the function.
+                // The system will try to add the service bus message back to the queue and will try to
+                // send the notification again. For now, we will store the current state as "Failed" in
+                // the respository, but it should not send a message to the data queue because we do not
+                // want to count a failure when the next attempt may succeed. If the system tries to send
+                // the notification repeatedly and reaches the dead letter maximum number of attempts,
+                // then the system should send a "Failed" message to the data queue. In this case, the
+                // the status code will not be Continue.
+                updatedSentNotificationDataEntity.DeliveryStatus = SentNotificationDataEntity.Continued;
+                sendDataQueueMessage = false;
+            }
+            else
+            {
+                updatedSentNotificationDataEntity.DeliveryStatus = SentNotificationDataEntity.Failed;
+                dataQueueMessageContent.ResultType = DataQueueResultType.Failed;
+            }
+
+            var sendDataQueueMessageTask = Task.CompletedTask;
+
+            if (sendDataQueueMessage)
```

Comment:
Change to use ternary operator instead of if.
sendDataQueueMessageTask = sendDataQueueMessage ? ... : Task.CompletedTask